### PR TITLE
Fix jdk17 javadoc errors

### PR DESCRIPTION
**Pull request by pshipton**
Fixes https://github.com/eclipse/openj9/issues/12132

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8,jdknext

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8,jdknext

File: jcl/src/java.management/share/classes/java/lang/management/PlatformLoggingMXBean.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
  * </li>
  * <li>Using a javax.management.MBeanServerConnection.</li>
  * <li>Obtaining a proxy MXBean from the static
- * {@link ManagementFactory#newPlatformMXBeanProxy} method, passing in
+ * {@link java.lang.management.ManagementFactory#newPlatformMXBeanProxy} method, passing in
```
Comment (keithc-ca) [inline jcl/src/java.management/share/classes/java/lang/management/PlatformLoggingMXBean.java:40]: The link on line 36 needs to be fixed as well:
` {@link java.lang.management.ManagementFactory#getPlatformMXBean}`

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 2]
```diff
@@ -2924,9 +2924,10 @@ public String getTypeName() {
  * 		Repeatable annotations are annotated with Repeatable annotation which tells the 
  * 		container annotation for this repeatable annotation.
  * 		Example = 
- * 
+ * 		<code>
  * 		@interface ContainerAnnotation {RepeatableAnn[] value();}
  * 		@Repeatable(ContainerAnnotation.class)
+ * 		</code>
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/Class.java:2930]: This documentation needs more formatting, as is, it is presented as a single paragraph:
> Terms used for annotations : Repeatable Annotation : An annotation which can be used more than once for the same class declaration. Repeatable annotations are annotated with Repeatable annotation which tells the container annotation for this repeatable annotation. Example =

It need not be addressed in this pull request.

Using `<code>...</code>` doesn't seem to work, but this does:
```
 * 			<pre>{@code
 * 		 	@interface ContainerAnnotation {RepeatableAnn[] value();}
 * 			@Repeatable(ContainerAnnotation.class)
 * 			}</pre>
```

File: jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/image/ImageFactory.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
  *
  * e.g.
  *
- * <pre>
+ * <code>
  * Image image;
  * try {
  *   Class<?> factoryClass = Class.forName(&quot;com.ibm.dtfj.image.j9.ImageFactory&quot;);
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/image/ImageFactory.java:43]: This should use `<pre>{@code ... }</pre>`.

File: jcl/src/jdk.management/share/classes/com/sun/management/GarbageCollectionNotificationInfo.java [hunk 4]
```diff
@@ -176,10 +176,10 @@ public GcInfo getGcInfo() {
 	 * given {@code CompositeData}.
 	 * The given {@code CompositeData} must contain the following attributes:
 	 * <blockquote>
-	 * <table border summary="">
+	 * <table border=1>
```
Comment (keithc-ca) [inline jcl/src/jdk.management/share/classes/com/sun/management/GarbageCollectionNotificationInfo.java:179]: I see this error also:
```
GarbageCollectionNotificationInfo.java:199: error: no caption for table
     * </table>
       ^
```

File: jcl/src/jdk.management/share/classes/com/ibm/lang/management/MemoryPoolMXBean.java [hunk 5]
```diff
@@ -94,7 +94,7 @@
  *	<li><b>balanced-old</b>
  * </ul>
  * 
- * <h3>4. Collection Usage Threshold ({@link java.lang.management.MemoryPoolMXBean#getCollectionUsageThreshold}, {@link java.lang.management.MemoryPoolMXBean#setCollectionUsageThreshold}, {@link java.lang.management.MemoryPoolMXBean#isCollectionUsageThresholdExceeded})</h3>
+ * <h2>4. Collection Usage Threshold ({@link java.lang.management.MemoryPoolMXBean#getCollectionUsageThreshold}, {@link java.lang.management.MemoryPoolMXBean#setCollectionUsageThreshold}, {@link java.lang.management.MemoryPoolMXBean#isCollectionUsageThresholdExceeded})</h2>
```
Comment (keithc-ca) [inline jcl/src/jdk.management/share/classes/com/ibm/lang/management/MemoryPoolMXBean.java:97]: This needs more work:
```
MemoryPoolMXBean.java:128: error: unexpected heading used: <H2>, compared to implicit preceding heading: <H3>
     * <h2>MBeanServer access:</h2>
       ^
```

File: jcl/src/jdk.management/share/classes/com/sun/management/GcInfo.java [hunk 6]
```diff
@@ -189,10 +189,10 @@ public long getDuration() {
 	 * all the following attributes:
 	 *
 	 * <blockquote>
-	 * <table border summary="">
+	 * <table border=1>
```
Comment (keithc-ca) [inline jcl/src/jdk.management/share/classes/com/sun/management/GcInfo.java:192]:
```
GcInfo.java:216: error: no caption for table
     * </table>
       ^
```

File: jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassUtilities.java [hunk 7]
```diff
@@ -177,7 +176,7 @@
 	 * @throws		IllegalStateException
 	 * 					If shared classes is disabled for this JVM (that is -Xshareclasses:none is present).
 	 * @throws		IllegalArgumentException
-	 * 					If <code>cacheType<code> is not a valid value.
+	 * 					If <code>cacheType</code> is not a valid value.
```
Comment (keithc-ca) [inline jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassUtilities.java:179]: We should prefer `{@code ...}` over `<code>...</code>`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Besides the specific comments, there's at least one other failure:
```
SharedDataHelper.java:29: error: heading used out of sequence: <H3>, compared to implicit preceding heading: <H1>
 * <h3>Description</h3>
   ^
```

File: jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassUtilities.java (hunk 7 above)
Comment (pshipton) [reply jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassUtilities.java:179]: I'm not keen to change all the places that use `<code>` just to continue to compile jdk17. I suggest this should remain as `<code>` to be consistent with the rest of the file, and changing to use `{@code ...}` should be done everywhere in a different PR.

Comment (keithc-ca) [reply jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassUtilities.java:179]: Actually, using `{@code` (where the closing `}`) is not on the same line breaks jdk11. :-(
We'll need to escape things like `@` (as `{@literal @}` or `&#x40;`), `<` (as `&lt;`), etc.

File: jcl/src/java.base/share/classes/java/lang/Class.java (hunk 2 above)
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/Class.java:2930]: Correction: we want `<pre>` and `</pre`> but we can't use `{@code` where the closing `}` is not on the same line without breaking earlier versions. This should be:
```
 * 		<pre><code>
 *	 	{@literal @}interface ContainerAnnotation {RepeatableAnn[] value();}
 *		{@literal @}Repeatable(ContainerAnnotation.class)
 *		</code></pre>
```

**Comment by pshipton** [comment]
I believe I've addressed all the review comments.

**Comment by pshipton** [comment]
I didn't see the new review comments, so there is more work to do.

**Comment by keithc-ca** [comment]
Errors building jdk11:
```
jdk11/build/normal/support/j9jcl/java.base/share/classes/java/lang/Class.java:2729: error: self-closing element not allowed
 * <br/>
   ^
jdk11/build/normal/support/j9jcl/java.base/share/classes/java/lang/Class.java:2730: error: self-closing element not allowed
 * Terms used for annotations :<br/><br/>
                               ^
jdk11/build/normal/support/j9jcl/java.base/share/classes/java/lang/Class.java:2730: error: self-closing element not allowed
 * Terms used for annotations :<br/><br/>
                                    ^
jdk11/build/normal/support/j9jcl/java.base/share/classes/java/lang/Class.java:2736: error: unterminated inline tag
 * 		<pre>{@code
   		     ^
```
Errors (and warnings) building jdk17:
```
/home/keithc/space/openj9/jdk17/build/normal/support/j9jcl/java.management/share/classes/java/lang/management/PlatformLoggingMXBean.java:35: error: reference not found
 * <li>Invoking the static {@link java.lang.management.ManagementFactory#getPlatformMXBean()} method.
                                  ^
/home/keithc/space/openj9/jdk17/build/normal/support/j9jcl/jdk.management/share/classes/com/sun/management/GarbageCollectionNotificationInfo.java:179: warning: empty <caption> tag
         * <caption></caption>
                    ^
/home/keithc/space/openj9/jdk17/build/normal/support/j9jcl/jdk.management/share/classes/com/sun/management/GcInfo.java:192: warning: empty <caption> tag
         * <caption></caption>
                    ^
/home/keithc/space/openj9/jdk17/build/normal/support/j9jcl/jdk.management/share/classes/com/ibm/lang/management/MemoryPoolMXBean.java:128: error: unexpected heading used: <H3>, compared to implicit preceding heading: <H3>
         * <h3>MBeanServer access:</h3>
           ^
```
The empty captions are added in this change: we should not be adding to the list of warnings.

**Comment by keithc-ca** [comment]
Testing underway in ibmruntimes/openj9-openjdk-jdk#294.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**